### PR TITLE
Read On Chain Current Supply

### PR DESCRIPTION
**Pull request by controtie**
# **Pull Request Template**

## **Type of Change**

*Select a category that relates to the content of your pull request (choose one).*

- [ ] Bug fix
- [x] New feature
- [ ] Documentation
- [ ] Code Refactoring
- [ ] Update of code styling

&nbsp;

## **Does this request include a breaking change?**

*Is there any potential for this change to crash Index UI or make it unusable?*

- [ ] Yes
- [x] No

&nbsp;

## **Description**

* Currently indexcoop.com calculates current supply by dividing market cap by unit price
* This PR reads current supply from the contract directly on chain

*If you are resolving an issue, please link the corresponding issue number.*

&nbsp;

**Summary of Changes:**
* Fetch DPI and MVI current supplies on chain
* Display current supplies on-chain data on product pages
* Display '--' when current supplies are unavailable

**Fixes:** Issue #

&nbsp;

## **Test Data or Screenshots**
image

&nbsp;

## **Checklist**

*Please review and confirm each of the following before creating your pull request.*

- [x] I have reviewed the [Contribution Guidelines](https://github.com/SetProtocol/index-ui/blob/master/CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have checked my code and corrected any errors.
- [x] I have commented my code or used an effective naming scheme.
- [x] I have updated my repository to match the master at `SetProtocol/index-ui`.
- [x] I have included test data or screenshots that prove my fix is effective or that my feature works.

&nbsp;

*If all of the above are **true**, then you may mark your pull request as **ready for review**.*

### CODE REVIEW

File: src/components/ProductPage/ProductMetaData.tsx [hunk 1]
```diff
@@ -73,10 +72,7 @@ const ProductMetaData: React.FC<ProductMetaDataProps> = ({ tokenData }) => {
         <StyledStat>
           <StyledStatTitle>Current Supply</StyledStatTitle>
           <StyledStatMetric>
-            {formatCurrentSupply(
-              tokenData.latestMarketCap,
-              tokenData.latestPrice
-            )}
+            {formattedCurrentySupply !== '0' ? formattedCurrentySupply : '--'}
```
Comment (controtie) [inline src/components/ProductPage/ProductMetaData.tsx:75]: This could be better. Open to suggestions

Comment (0xModene) [reply src/components/ProductPage/ProductMetaData.tsx:75]: Perhaps put that logic in the formatter, then this is a pure function read

```
const formatCurrentSupply = (currentSupply: BigNumber | undefined) =>
    currentSupply ? numeral(currentSupply).format('0,0')  :  '--'
```

File: src/components/ProductPage/ProductTokenStats.tsx [hunk 2]
```diff
@@ -41,6 +42,10 @@ const ProductTokenStats: React.FC<ProductTokenStatsProps> = ({
     </StyledStat>
   )
 
+  const formattedCurrentSupply = numeral(
```
Comment (0xModene) [inline src/components/ProductPage/ProductTokenStats.tsx:45]: Replicate the ProductMetaData logic here

**Review by 0xModene — CHANGES_REQUESTED** [review]
Mostly good, small change suggested on the stat/meta data formatter

**Review by 0xModene — APPROVED** [review]
gang gang

**Pull request merged**